up-transport-vsomeip/src/transport.rs

```diff
@@ -0,0 +1,57 @@
+/********************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use crate::UPTransportVsomeip;
+use async_trait::async_trait;
+use std::sync::Arc;
+use up_rust::{UCode, UListener, UMessage, UStatus, UTransport, UUri};
+
+#[async_trait]
+impl UTransport for UPTransportVsomeip {
+    async fn send(&self, message: UMessage) -> Result<(), UStatus> {
+        self.transport_inner.send(message).await
+    }
+
+    async fn register_listener(
```

Comment:
given that this implementation does not do anything but forwarding to transport_inner, I wonder why we need both ...

Reply:
Perhaps premature decoupling? The idea being that we could in the future implement the transport_inner differently, for example without using the handle / engine idea with passing commands into the engine.

Looking for perhaps some architecture / structuring advice here.

Reply:
Would like to hear some thoughts here tomorrow if you have time.

Reply:
Ping on this one too.

Reply:
Leaving this as-is for now.

I could be convinced to collapse this in a future revision.